src/core/cm/imagemanager/itf/imagemanager.hpp

```diff
@@ -0,0 +1,159 @@
+    DecryptInfo               mDecryptInfo;
+    SignInfo                  mSignInfo;
+
+    /**
+     * Compares update image info.
+     *
+     * @param other info to compare with.
+     * @return bool.
+     */
+    bool operator==(const UpdateImageInfo& other) const
+    {
+        return mImage == other.mImage && mURL == other.mURL && mSHA256 == other.mSHA256 && mSize == other.mSize
+            && mDecryptInfo == other.mDecryptInfo && mSignInfo == other.mSignInfo;
+    }
+
+    /**
+     * Compares update item info.
+     */
+    bool operator!=(const UpdateImageInfo& other) const { return !operator==(other); }
+};
+
+/**
+ * Update item info.
+ */
+struct UpdateItemInfo {
+    StaticString<cURLLen>                              mURN;
+    StaticString<cVersionLen>                          mVersion;
+    StaticArray<InstallImageInfo, cMaxNumUpdateImages> mImages;
+
+    /**
+     * Compares update item info.
+     *
+     * @param other object to compare with.
+     * @return bool.
+     */
+    bool operator==(const UpdateItemInfo& other) const
+    {
+        return mURN == other.mURN && mVersion == other.mVersion && mImages == other.mImages;
+    }
+
+    /**
+     * Compares install item info.
+     *
+     * @param other object to compare with.
+     * @return bool.
+     */
+    bool operator!=(const UpdateItemInfo& other) const { return !operator==(other); }
+};
+
+/**
+ * Update item status.
+ */
+struct UpdateItemStatus {
+    StaticString<cURNLen>                               mURN;
+    StaticString<cVersionLen>                           mVersion;
+    StaticArray<UpdateImageStatus, cMaxNumUpdateImages> mStatuses;
+
+    /**
+     * Compares update item status.
+     *
+     * @param other update item status to compare with.
+     * @return bool.
+     */
+    bool operator==(const UpdateItemStatus& other) const
+    {
+        return mURN == other.mURN && mVersion == other.mVersion && mStatuses == other.mStatuses;
+    }
+
+    /**
+     * @brief Compares update item status.
+     *
+     * @param other update item status to compare with.
+     * @return bool.
+     */
+    bool operator!=(const UpdateItemStatus& other) const { return !operator==(other); }
+};
+
+/**
+ * Interface that manages update items images.
+ */
+class ImageManagerItf {
+public:
+    /**
+     * Destructor.
+     */
+    virtual ~ImageManagerItf() = default;
+
+    /**
+     * Retrieves update items statuses.
+     *
+     * @param[out] statuses list of update items statuses.
+     * @return Error.
+     */
+    virtual Error GetUpdateItemsStatuses(Array<UpdateItemStatus>& statuses) = 0;
+
+    /**
+     *
+     * Installs update item.
+     *
+     * @param itemInfo update item info.
+     * @param certificates list of certificates.
+     * @param certificateChains list of certificate chains.
+     * @param[out] status update item status.
+     * @return Error.
+     */
+    virtual Error InstallUpdateItem(const UpdateItemInfo& itemInfo,
```

Comment:
According to the definition of UpdateItemInfo, it does not include a field that indicates what type of item or image to install.

Reply:
It is part of URN, cloud team should provide us how to parse URN.